public/app/plugins/datasource/prometheus/datasource.test.ts

```diff
@@ -97,6 +98,72 @@ describe('PrometheusDatasource', () => {
         expect(response[0].state).toBe(LoadingState.Done);
       });
     });
+
+    describe('step parameter', () => {
+      const to = dateTime();
+      const from = dateTime(to).subtract(6, 'hours');
+      const sixHoursRange = { from, to, raw: { from: 'now-6h', to: 'now' } };
+
+      describe('When set to exact', () => {
+        it('should return an error if interval is modified', async () => {
+          const request = createDataRequest([{ interval: '1s', stepMode: 'exact' }], {
+            range: sixHoursRange,
+          });
+
+          await expect(ds.query({ ...request, app: CoreApp.Dashboard })).toEmitValuesWith((response) => {
```

Comment:
Could we also assert that the error message is correct? Or assert the correct safe interval was applied?

Reply:
> Could we also assert that the error message is correct?

how would you go about that? I also wanted to test it, but at that point I was just copying and pasting the error message in the test, which didn't seem very robust

> Or assert the correct safe interval was applied?

https://github.com/grafana/grafana/blob/46d0a72cf526e6ce3db2b1232e6b5b17bdb71f2a/public/app/plugins/datasource/prometheus/datasource.test.ts#L1187 has already some tests for the safe interval.